UnitTests/GTMSessionFetcherChunkedUploadTest.m

```diff
@@ -809,6 +809,201 @@ - (void)testBigFileURLSingleChunkedUploadFetchRetry {
   [self removeTemporaryFileURL:bigFileURL];
 }
 
+- (void)testBigFileURLSingleChunkedUploadFetchRetryWithOffset {
+  CREATE_START_STOP_NOTIFICATION_EXPECTATIONS(1, 1);
+  // Like testBigFileURLSingleChunkedUploadFetch, but the initial request will fail
+  // with HTTP 503, triggering a retry.
+  FetcherNotificationsCounter *fnctr = [[FetcherNotificationsCounter alloc] init];
+
+  NSURL *bigFileURL = [self bigFileToUploadURLWithBaseName:NSStringFromSelector(_cmd)];
+
+  NSMutableURLRequest *request = [self validUploadFileRequest];
+
+  NSURL *originalURL = request.URL;
+  NSString *failureURL = [originalURL.absoluteString stringByAppendingString:@"?status=503"];
+  request.URL = [NSURL URLWithString:failureURL];
+
+  GTMSessionUploadFetcher *fetcher =
+      [GTMSessionUploadFetcher uploadFetcherWithRequest:request
+                                         uploadMIMEType:@"text/plain"
+                                              chunkSize:kGTMSessionUploadFetcherStandardChunkSize
+                                         fetcherService:_service];
+  fetcher.uploadFileURL = bigFileURL;
+  fetcher.useBackgroundSession = NO;
+  fetcher.allowLocalhostRequest = YES;
+
+  BOOL (^shouldRetryUpload)(GTMSessionUploadFetcher *, BOOL, NSError *) =
+      ^BOOL(GTMSessionUploadFetcher *blockFetcher, BOOL suggestedWillRetry, NSError *error) {
+        // Change this fetch's request to have the original, non-failure status URL.
+        // This will make the retry succeed.
+        return suggestedWillRetry;  // do the retry fetch; it should succeed now
+      };
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-retain-cycles"
```

Comment:
Can we capture the fetcher weakly to avoid the `-Warc-retain-cycles` warning? I would prefer we avoid introducing retain cycles in the tests.